### PR TITLE
Add accept test

### PR DESCRIPTION
**Pull request by kyriesent**
Add the commands
- `hubot accept me`
- `hubot accept logs <test-id>`

### CODE REVIEW

File: bin/accept-test.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+
+cd /mnt/subtree-test/services/monolith
+git pull --depth=1
+npm install
+sudo bin/acceptTest -h https://seelio:parkour@stage.seelio.com
```
Comment (kyriesent) [inline bin/accept-test.sh]: This shell script gets executed on the server that runs the acceptance tests ( currently the build server )

File: bin/run-accept-tests [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+chmod 600 .ssh/id_rsa
+
+touch .ssh/config
+grep -q -F 'LogLevel ERROR' .ssh/config || sed -i '1i LogLevel ERROR' .ssh/config
+grep -q -F "Host ec2-52-5-152-200.compute-1.amazonaws.com" .ssh/config || echo "Host ec2-52-5-152-200.compute-1.amazonaws.com
+StrictHostKeyChecking no
+" >> .ssh/config
+
+ssh -oBatchMode=yes -oConnectTimeout=3 -i .ssh/id_rsa -F .ssh/config -o UserKnownHostsFile=.ssh/known_hosts -T ubuntu@ec2-52-5-152-200.compute-1.amazonaws.com < ./accept-test.sh
+rm -Rf .ssh
```
Comment (kyriesent) [inline bin/run-accept-tests]: This script is executed on the same machine that hubot runs on. It ssh's into a remote machine (currently the build server) and runs the accept-test.sh script on it.

File: scripts/accept-test.coffee [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  @id = Math.ceil(Math.random() * 100000)
+  @logs = ""
+
+Test::log = (text) ->
+  @logs += text + '\n'
+
+module.exports = (robot) ->
+  tests = {}
+
+  robot.respond /accept me/i, (msg) ->
```
Comment (kyriesent) [inline scripts/accept-test.coffee]: `hubot accept me` will run the acceptance tests against staging

File: scripts/accept-test.coffee [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+      msg.send "Acceptance testing is finished! (Test ID: #{test.id})"
+
+
+    test = new Test child
+    
+    tests[test.id] = test
+    msg.send "Running acceptance tests against UA server! I will inform you when I am finished. (Test ID: #{test.id})"
+
+  robot.respond /accept logs (.*)/i, (msg) ->
```
Comment (kyriesent) [inline scripts/accept-test.coffee]: `hubot accept logs <test-id>` will spit back the logs from the test

File: scripts/accept-test.coffee [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    child.on 'error', (err) ->
+      if err
+        msg.send "Oh noes! Test ##{test.id} failed! Error: #{err.message}"
+        return
+
+    child.on 'exit', (code, signal) ->
+      # return msg.send "Deploying #{app} canceled! (Deployment ID: #{deployment.id})"  if signal is 'SIGTERM'
+      # return msg.send "Oh noes! Deployment ##{deployment.id} failed! Check the build logs. (Exit code: #{code})" if code? and code isnt 0
+
+      msg.send "Acceptance testing is finished! (Test ID: #{test.id})"
```
Comment (kyriesent) [inline scripts/accept-test.coffee]: This doesn't actually tell you if the test passed or not yet. That will come in a subsequent PR.

**Comment by bfla** [comment]
:+1: This is gonna be rad!

**Pull request merged**